packages/builder/src/components/backend/DatasourceNavigator/popovers/EditQueryPopover.svelte

```diff
@@ -11,13 +11,22 @@
     await queries.delete(query)
     notifications.success("Query deleted")
   }
+
+  async function duplicateQuery() {
+    try {
+      await queries.duplicate(query, queries.save)
```

Comment:
We could refactor how the actions in the `queries` store are defined, so that they can reference each other. This would remove the need to pass in the save function as a param here.

Basically defining the queries store something like this:
```js
...
const { subscribe, set, update } = store
const actions = {
  ...
  save: () => { ... },
  duplicate: () => {
    ...
    actions.save()
    ...
  }
}
return {
  subscribe,
  set,
  update,
  ...actions
}
```

It would just feel more natural to me to be able to call `queries.duplicate(query)` without having to pass in that second param, and I assume the current inability to reference `save` from `duplicate` was the main motivator for that. If there's another reason that I'm not thinking of at the moment though then let me know.

Reply:
Yep spot on, I wasn't sure how to reference another store action short of pulling it out to an outer function which didn't feel right to do for one and not the others. Your suggestion is great and works like a charm, I didn't think an inner object function could reference the variable it was being defined in 🤯 

Added here https://github.com/Budibase/budibase/pull/3702/commits/3bd93f275585bc0560198b347c499c5f6b1ff536